lualib/resty/kong/request.lua

```diff
@@ -0,0 +1,35 @@
+local ffi = require "ffi"
+local base = require "resty.core.base"
+local get_request = base.get_request
+
+local C = ffi.C
+local ffi_new = ffi.new
+local ffi_str = ffi.string
+
+local subsystem = ngx.config.subsystem
+
+local get_header
+if subsystem == "http" then
+    ffi.cdef[[
+        ngx_str_t *ngx_http_lua_kong_ffi_request_get_header(
+            ngx_http_request_t *r, ngx_str_t name, size_t search_limit);
+    ]]
+
+    local DEFAULT_HEADER_LIMIT = 100
+
+    local name_str = ffi_new("ngx_str_t")
+    get_header = function (name, limit)
```

Comment:
we should do sanity check on `name` and `limit`.

Reply:
done